// @srclaunch/types/dist/communications/chat.js
export var ChatMessageStatus;
(function (ChatMessageStatus) {
    ChatMessageStatus["Delivered"] = "delivered";
    ChatMessageStatus["Read"] = "read";
    ChatMessageStatus["Sending"] = "sending";
    ChatMessageStatus["Sent"] = "sent";
})(ChatMessageStatus || (ChatMessageStatus = {}));
export var ChatMessageType;
(function (ChatMessageType) {
    ChatMessageType["Audio"] = "audio";
    ChatMessageType["File"] = "file";
    ChatMessageType["Image"] = "image";
    ChatMessageType["Text"] = "text";
    ChatMessageType["Video"] = "video";
})(ChatMessageType || (ChatMessageType = {}));
export var ChatMessageAttachmentType;
(function (ChatMessageAttachmentType) {
    ChatMessageAttachmentType["Audio"] = "audio";
    ChatMessageAttachmentType["File"] = "file";
    ChatMessageAttachmentType["Image"] = "image";
    ChatMessageAttachmentType["Video"] = "video";
})(ChatMessageAttachmentType || (ChatMessageAttachmentType = {}));
export var ChatMessageReactionType;
(function (ChatMessageReactionType) {
    ChatMessageReactionType["Angry"] = "angry";
    ChatMessageReactionType["Laugh"] = "laugh";
    ChatMessageReactionType["Like"] = "like";
    ChatMessageReactionType["Love"] = "love";
    ChatMessageReactionType["Sad"] = "sad";
    ChatMessageReactionType["Wow"] = "wow";
    ChatMessageReactionType["Wink"] = "wink";
    ChatMessageReactionType["Yay"] = "yay";
})(ChatMessageReactionType || (ChatMessageReactionType = {}));
