// @srclaunch/types/dist/applab/rules/condition/index.js
export var Condition;
(function (Condition) {
    Condition["Contains"] = "contains";
    Condition["HasCharacterCount"] = "has-character-count";
    Condition["HasNumberCount"] = "has-number-count";
    Condition["HasLetterCount"] = "has-letter-count";
    Condition["HasLowercaseCount"] = "has-lowercase-count";
    Condition["HasSpacesCount"] = "has-spaces-count";
    Condition["HasSymbolCount"] = "has-symbol-count";
    Condition["HasUppercaseCount"] = "has-uppercase-count";
    Condition["IsAfter"] = "is-after";
    Condition["IsAfterOrEqual"] = "is-after-or-equal";
    Condition["IsAirport"] = "is-airport";
    Condition["IsAlpha"] = "is-alpha";
    Condition["IsAlphanumeric"] = "is-alphanumeric";
    Condition["IsAlgorithmHash"] = "is-algorithm-hash";
    Condition["IsAscii"] = "is-ascii";
    Condition["IsBase64"] = "is-base-64";
    Condition["IsBefore"] = "is-before";
    Condition["IsBeforeOrAfter"] = "is-before-or-after";
    Condition["IsBeforeOrEqual"] = "is-before-or-equal";
    Condition["IsBetween"] = "is-between";
    Condition["IsBIC"] = "is-bic";
    Condition["IsBitcoinAddress"] = "is-bitcoin-address";
    Condition["IsBoolean"] = "is-boolean";
    Condition["IsColor"] = "is-color";
    Condition["IsComplexEnough"] = "is-complex-enough";
    Condition["IsCountry"] = "is-country";
    Condition["IsCreditCard"] = "is-credit-card";
    Condition["IsCurrency"] = "is-currency";
    Condition["IsDataURI"] = "is-data-uri";
    Condition["IsDate"] = "is-date";
    Condition["IsDateRange"] = "is-date-range";
    Condition["IsDateTime"] = "is-date-time";
    Condition["IsDayOfMonth"] = "is-day-of-month";
    Condition["IsDecimal"] = "is-decimal";
    Condition["IsDivisibleBy"] = "is-divisible-by";
    Condition["IsDomainName"] = "is-domain-name";
    Condition["IsEmailAddress"] = "is-email-address";
    Condition["IsEthereumAddress"] = "is-ethereum-address";
    Condition["IsEAN"] = "is-ean";
    Condition["IsEIN"] = "is-ein";
    Condition["IsEqual"] = "is-equal";
    Condition["IsEvenNumber"] = "is-even-number";
    Condition["IsFloat"] = "is-float";
    Condition["IsIBAN"] = "is-iban";
    Condition["IsGreaterThan"] = "greater-than";
    Condition["IsGreaterThanOrEqual"] = "greater-than-or-equal";
    Condition["IsHSLColor"] = "is-hsl-color";
    Condition["IsHexColor"] = "is-hex-color";
    Condition["IsHexadecimal"] = "is-hexadecimal";
    Condition["IsIdentityCardCode"] = "is-identity-card-code";
    Condition["IsIMEI"] = "is-imei";
    Condition["IsInIPAddressRange"] = "is-in-ip-address-range";
    Condition["IsInList"] = "is-in-list";
    Condition["IsInTheLast"] = "is-in-the-last";
    Condition["IsInteger"] = "is-integer";
    Condition["IsIPAddress"] = "is-ip-address";
    Condition["IsIPAddressRange"] = "is-ip-address-range";
    Condition["IsISBN"] = "is-isbn";
    Condition["IsISIN"] = "is-isin";
    Condition["IsISMN"] = "is-ismn";
    Condition["IsISRC"] = "is-isrc";
    Condition["IsISSN"] = "is-issn";
    Condition["IsISO4217"] = "is-iso-4217";
    Condition["IsISO8601"] = "is-iso-8601";
    Condition["IsISO31661Alpha2"] = "is-iso-31661-alpha-2";
    Condition["IsISO31661Alpha3"] = "is-iso-31661-alpha-3";
    Condition["IsJSON"] = "is-json";
    Condition["IsLanguage"] = "is-language";
    Condition["IsLatitude"] = "is-latitude";
    Condition["IsLongitude"] = "is-longitude";
    Condition["IsLengthEqual"] = "is-length-equal";
    Condition["IsLengthGreaterThan"] = "is-length-greater-than";
    Condition["IsLengthGreaterThanOrEqual"] = "is-length-great-than-or-equal";
    Condition["IsLengthLessThan"] = "is-length-less-than";
    Condition["IsLengthLessThanOrEqual"] = "is-length-less-than-or-equal";
    Condition["IsLessThan"] = "less-than";
    Condition["IsLessThanOrEqual"] = "less-than-or-equal";
    Condition["IsLicensePlateNumber"] = "is-license-plate-number";
    Condition["IsLowercase"] = "is-lowercase";
    Condition["IsOctal"] = "is-octal";
    Condition["IsMACAddress"] = "is-mac-address";
    Condition["IsMD5"] = "is-md5";
    Condition["IsMagnetURI"] = "is-magnet-uri";
    Condition["IsMarkdown"] = "is-markdown";
    Condition["IsMimeType"] = "is-mime-type";
    Condition["IsMonth"] = "is-month";
    Condition["IsNegativeNumber"] = "is-negative-number";
    Condition["IsNotDate"] = "is-not-date";
    Condition["IsNotEqual"] = "is-not-equal";
    Condition["IsNotInIPAddressRange"] = "is-not-in-ip-address-range";
    Condition["IsNotInList"] = "is-not-in-list";
    Condition["IsNotNull"] = "is-not-null";
    Condition["IsNotRegexMatch"] = "is-not-regex-match";
    Condition["IsNotToday"] = "is-not-today";
    Condition["IsNumber"] = "is-number";
    Condition["IsNumeric"] = "is-numeric";
    Condition["IsOddNumber"] = "is-odd-number";
    Condition["IsPassportNumber"] = "is-passport-number";
    Condition["IsPhoneNumber"] = "is-phone-number";
    Condition["IsPort"] = "is-port";
    Condition["IsPositiveNumber"] = "is-positive-number";
    Condition["IsPostalCode"] = "is-postal-code";
    Condition["IsProvince"] = "is-province";
    Condition["IsRGBColor"] = "is-rgb-color";
    Condition["IsRegexMatch"] = "is-regex-match";
    Condition["IsRequired"] = "is-required";
    Condition["IsSemanticVersion"] = "is-semantic-version";
    Condition["IsSlug"] = "is-slug";
    Condition["IsSSN"] = "is-ssn";
    Condition["IsState"] = "is-state";
    Condition["IsStreetAddress"] = "is-street-address";
    Condition["IsString"] = "is-string";
    Condition["IsStrongPassword"] = "is-strong-password";
    Condition["IsTags"] = "is-tags";
    Condition["IsTaxIDNumber"] = "is-tax-id-number";
    Condition["IsThisMonth"] = "is-this-month";
    Condition["IsThisQuarter"] = "is-this-quarter";
    Condition["IsThisWeek"] = "is-this-week";
    Condition["IsThisWeekend"] = "is-this-weekend";
    Condition["IsThisYear"] = "is-this-year";
    Condition["IsTime"] = "is-time";
    Condition["IsTimeOfDay"] = "is-time-of-day";
    Condition["IsTimeRange"] = "is-time-range";
    Condition["IsToday"] = "is-today";
    Condition["IsURL"] = "is-url";
    Condition["IsUUID"] = "is-uuid";
    Condition["IsUppercase"] = "is-uppercase";
    Condition["IsUsernameAvailable"] = "is-username-available";
    Condition["IsValidStreetAddress"] = "is-valid-street-address";
    Condition["IsVATIDNumber"] = "is-vat-id-number";
    Condition["IsWeekday"] = "is-weekday";
    Condition["IsWeekend"] = "is-weekend";
    Condition["IsYear"] = "is-year";
})(Condition || (Condition = {}));
export var AuthenticationCondition;
(function (AuthenticationCondition) {
    AuthenticationCondition["IsAuthenticated"] = "is-authenticated";
    AuthenticationCondition["IsNotAuthenticated"] = "is-not-authenticated";
    AuthenticationCondition["IsUsernameAvailable"] = "is-username-available";
    AuthenticationCondition["PasswordMismatch"] = "password-mismatch";
})(AuthenticationCondition || (AuthenticationCondition = {}));
