// @srclaunch/types/dist/applab/rules/condition/primitive/application.js
import { Condition } from '../index';
export var AlgorithmHashCondition;
(function (AlgorithmHashCondition) {
    AlgorithmHashCondition["IsAlgorithmHash"] = "is-algorithm-hash";
    AlgorithmHashCondition["IsEqual"] = "is-equal";
    AlgorithmHashCondition["IsInList"] = "is-in-list";
    AlgorithmHashCondition["IsNotEqual"] = "is-not-equal";
    AlgorithmHashCondition["IsNotInList"] = "is-not-in-list";
    AlgorithmHashCondition["IsNotNull"] = "is-not-null";
    AlgorithmHashCondition["IsString"] = "is-string";
})(AlgorithmHashCondition || (AlgorithmHashCondition = {}));
export var SemanticVersionCondition;
(function (SemanticVersionCondition) {
    SemanticVersionCondition["IsEqual"] = "is-equal";
    SemanticVersionCondition["IsInList"] = "is-in-list";
    SemanticVersionCondition["IsNotEqual"] = "is-not-equal";
    SemanticVersionCondition["IsNotInList"] = "is-not-in-list";
    SemanticVersionCondition["IsNotNull"] = "is-not-null";
    SemanticVersionCondition["IsSemanticVersion"] = "is-semantic-version";
    SemanticVersionCondition["IsString"] = "is-string";
})(SemanticVersionCondition || (SemanticVersionCondition = {}));
export var UUIDCondition;
(function (UUIDCondition) {
    UUIDCondition["IsEqual"] = "is-equal";
    UUIDCondition["IsInList"] = "is-in-list";
    UUIDCondition["IsNotEqual"] = "is-not-equal";
    UUIDCondition["IsNotInList"] = "is-not-in-list";
    UUIDCondition["IsNotNull"] = "is-not-null";
    UUIDCondition["IsString"] = "is-string";
    UUIDCondition["IsUUID"] = "is-uuid";
})(UUIDCondition || (UUIDCondition = {}));
export var MD5Condition;
(function (MD5Condition) {
    MD5Condition["IsEqual"] = "is-equal";
    MD5Condition["IsInList"] = "is-in-list";
    MD5Condition["IsMD5"] = "is-md5";
    MD5Condition["IsNotEqual"] = "is-not-equal";
    MD5Condition["IsNotInList"] = "is-not-in-list";
    MD5Condition["IsNotNull"] = "is-not-null";
    MD5Condition["IsString"] = "is-string";
})(MD5Condition || (MD5Condition = {}));
