// @srclaunch/types/dist/applab/rules/condition/primitive/date.js
import { Condition } from '../index';
export var DateCondition;
(function (DateCondition) {
    DateCondition["IsAfter"] = "is-after";
    DateCondition["IsAfterOrEqual"] = "is-after-or-equal";
    DateCondition["IsBefore"] = "is-before";
    DateCondition["IsBeforeOrEqual"] = "is-before-or-equal";
    DateCondition["IsBetween"] = "is-between";
    DateCondition["IsDate"] = "is-date";
    DateCondition["IsEqual"] = "is-equal";
    DateCondition["IsNotDate"] = "is-not-date";
    DateCondition["IsNotEqual"] = "is-not-equal";
    DateCondition["IsNotNull"] = "is-not-null";
    DateCondition["IsNotToday"] = "is-not-today";
    DateCondition["IsThisWeek"] = "is-this-week";
    DateCondition["IsThisMonth"] = "is-this-month";
    DateCondition["IsThisQuarter"] = "is-this-quarter";
    DateCondition["IsThisYear"] = "is-this-year";
    DateCondition["IsToday"] = "is-today";
    DateCondition["IsWeekend"] = "is-weekend";
})(DateCondition || (DateCondition = {}));
export var DateRangeCondition;
(function (DateRangeCondition) {
    DateRangeCondition["IsAfter"] = "is-after";
    DateRangeCondition["IsBefore"] = "is-before";
    DateRangeCondition["IsBeforeOrAfter"] = "is-before-or-after";
    DateRangeCondition["IsBetween"] = "is-between";
    DateRangeCondition["IsDate"] = "is-date";
    DateRangeCondition["IsDateRange"] = "is-date-range";
    DateRangeCondition["IsEqual"] = "is-equal";
    DateRangeCondition["IsNotEqual"] = "is-not-equal";
    DateRangeCondition["IsNotNull"] = "is-not-null";
})(DateRangeCondition || (DateRangeCondition = {}));
export var DateTimeCondition;
(function (DateTimeCondition) {
    DateTimeCondition["IsAfter"] = "is-after";
    DateTimeCondition["IsAfterOrEqual"] = "is-after-or-equal";
    DateTimeCondition["IsBefore"] = "is-before";
    DateTimeCondition["IsBeforeOrEqual"] = "is-before-or-equal";
    DateTimeCondition["IsBetween"] = "is-between";
    DateTimeCondition["IsDate"] = "is-date";
    DateTimeCondition["IsEqual"] = "is-equal";
    DateTimeCondition["IsNotDate"] = "is-not-date";
    DateTimeCondition["IsNotEqual"] = "is-not-equal";
    DateTimeCondition["IsNotNull"] = "is-not-null";
    DateTimeCondition["IsNotToday"] = "is-not-today";
    DateTimeCondition["IsThisWeek"] = "is-this-week";
    DateTimeCondition["IsThisMonth"] = "is-this-month";
    DateTimeCondition["IsThisQuarter"] = "is-this-quarter";
    DateTimeCondition["IsThisYear"] = "is-this-year";
    DateTimeCondition["IsToday"] = "is-today";
    DateTimeCondition["IsWeekend"] = "is-weekend";
})(DateTimeCondition || (DateTimeCondition = {}));
export var DayOfMonthCondition;
(function (DayOfMonthCondition) {
    DayOfMonthCondition["IsAfter"] = "is-after";
    DayOfMonthCondition["IsAfterOrEqual"] = "is-after-or-equal";
    DayOfMonthCondition["IsBefore"] = "is-before";
    DayOfMonthCondition["IsBeforeOrEqual"] = "is-before-or-equal";
    DayOfMonthCondition["IsBetween"] = "is-between";
    DayOfMonthCondition["IsDayOfMonth"] = "is-day-of-month";
    DayOfMonthCondition["IsEvenNumber"] = "is-even-number";
    DayOfMonthCondition["IsEqual"] = "is-equal";
    DayOfMonthCondition["IsGreaterThan"] = "greater-than";
    DayOfMonthCondition["IsGreaterThanOrEqual"] = "greater-than-or-equal";
    DayOfMonthCondition["IsInteger"] = "is-integer";
    DayOfMonthCondition["IsLessThan"] = "less-than";
    DayOfMonthCondition["IsLessThanOrEqual"] = "less-than-or-equal";
    DayOfMonthCondition["IsNotEqual"] = "is-not-equal";
    DayOfMonthCondition["IsNotNull"] = "is-not-null";
    DayOfMonthCondition["IsNumber"] = "is-number";
    DayOfMonthCondition["IsOddNumber"] = "is-odd-number";
    DayOfMonthCondition["IsToday"] = "is-today";
    DayOfMonthCondition["IsWeekday"] = "is-weekday";
    DayOfMonthCondition["IsWeekend"] = "is-weekend";
})(DayOfMonthCondition || (DayOfMonthCondition = {}));
export var MonthCondition;
(function (MonthCondition) {
    MonthCondition["IsAfter"] = "is-after";
    MonthCondition["IsAfterOrEqual"] = "is-after-or-equal";
    MonthCondition["IsBefore"] = "is-before";
    MonthCondition["IsBeforeOrEqual"] = "is-before-or-equal";
    MonthCondition["IsBetween"] = "is-between";
    MonthCondition["IsEvenNumber"] = "is-even-number";
    MonthCondition["IsEqual"] = "is-equal";
    MonthCondition["IsGreaterThan"] = "greater-than";
    MonthCondition["IsGreaterThanOrEqual"] = "greater-than-or-equal";
    MonthCondition["IsInteger"] = "is-integer";
    MonthCondition["IsLessThan"] = "less-than";
    MonthCondition["IsLessThanOrEqual"] = "less-than-or-equal";
    MonthCondition["IsMonth"] = "is-month";
    MonthCondition["IsNotEqual"] = "is-not-equal";
    MonthCondition["IsNotNull"] = "is-not-null";
    MonthCondition["IsNumber"] = "is-number";
    MonthCondition["IsOddNumber"] = "is-odd-number";
    MonthCondition["IsThisMonth"] = "is-this-month";
})(MonthCondition || (MonthCondition = {}));
export var TimeCondition;
(function (TimeCondition) {
    TimeCondition["IsAfter"] = "is-after";
    TimeCondition["IsAfterOrEqual"] = "is-after-or-equal";
    TimeCondition["IsBefore"] = "is-before";
    TimeCondition["IsBeforeOrEqual"] = "is-before-or-equal";
    TimeCondition["IsBetween"] = "is-between";
    TimeCondition["IsEqual"] = "is-equal";
    TimeCondition["IsNotEqual"] = "is-not-equal";
    TimeCondition["IsNotNull"] = "is-not-null";
    TimeCondition["IsTime"] = "is-time";
})(TimeCondition || (TimeCondition = {}));
export var TimeRangeCondition;
(function (TimeRangeCondition) {
    TimeRangeCondition["IsAfter"] = "is-after";
    TimeRangeCondition["IsBefore"] = "is-before";
    TimeRangeCondition["IsBeforeOrAfter"] = "is-before-or-after";
    TimeRangeCondition["IsBetween"] = "is-between";
    TimeRangeCondition["IsTime"] = "is-time";
    TimeRangeCondition["IsEqual"] = "is-equal";
    TimeRangeCondition["IsNotEqual"] = "is-not-equal";
    TimeRangeCondition["IsNotNull"] = "is-not-null";
    TimeRangeCondition["IsTimeRange"] = "is-time-range";
})(TimeRangeCondition || (TimeRangeCondition = {}));
export var TimeOfDayCondition;
(function (TimeOfDayCondition) {
    TimeOfDayCondition["IsAfter"] = "is-after";
    TimeOfDayCondition["IsBefore"] = "is-before";
    TimeOfDayCondition["IsBeforeOrAfter"] = "is-before-or-after";
    TimeOfDayCondition["IsBetween"] = "is-between";
    TimeOfDayCondition["IsEqual"] = "is-equal";
    TimeOfDayCondition["IsInList"] = "is-in-list";
    TimeOfDayCondition["IsNotEqual"] = "is-not-equal";
    TimeOfDayCondition["IsNotInList"] = "is-not-in-list";
    TimeOfDayCondition["IsNotNull"] = "is-not-null";
    TimeOfDayCondition["IsTimeOfDay"] = "is-time-of-day";
    TimeOfDayCondition["IsTimeRange"] = "is-time-range";
})(TimeOfDayCondition || (TimeOfDayCondition = {}));
export var WeekdayCondition;
(function (WeekdayCondition) {
    WeekdayCondition["IsAfter"] = "is-after";
    WeekdayCondition["IsAfterOrEqual"] = "is-after-or-equal";
    WeekdayCondition["IsBefore"] = "is-before";
    WeekdayCondition["IsBeforeOrEqual"] = "is-before-or-equal";
    WeekdayCondition["IsBetween"] = "is-between";
    WeekdayCondition["IsEvenNumber"] = "is-even-number";
    WeekdayCondition["IsEqual"] = "is-equal";
    WeekdayCondition["IsGreaterThan"] = "greater-than";
    WeekdayCondition["IsGreaterThanOrEqual"] = "greater-than-or-equal";
    WeekdayCondition["IsLessThan"] = "less-than";
    WeekdayCondition["IsLessThanOrEqual"] = "less-than-or-equal";
    WeekdayCondition["IsNotEqual"] = "is-not-equal";
    WeekdayCondition["IsNotNull"] = "is-not-null";
    WeekdayCondition["IsNumber"] = "is-number";
    WeekdayCondition["IsOddNumber"] = "is-odd-number";
    WeekdayCondition["IsWeekday"] = "is-weekday";
    WeekdayCondition["IsWeekend"] = "is-weekend";
})(WeekdayCondition || (WeekdayCondition = {}));
export var YearCondition;
(function (YearCondition) {
    YearCondition["IsAfter"] = "is-after";
    YearCondition["IsAfterOrEqual"] = "is-after-or-equal";
    YearCondition["IsBefore"] = "is-before";
    YearCondition["IsBeforeOrEqual"] = "is-before-or-equal";
    YearCondition["IsBetween"] = "is-between";
    YearCondition["IsEvenNumber"] = "is-even-number";
    YearCondition["IsEqual"] = "is-equal";
    YearCondition["IsGreaterThan"] = "greater-than";
    YearCondition["IsGreaterThanOrEqual"] = "greater-than-or-equal";
    YearCondition["IsInteger"] = "is-integer";
    YearCondition["IsLessThan"] = "less-than";
    YearCondition["IsLessThanOrEqual"] = "less-than-or-equal";
    YearCondition["IsNotEqual"] = "is-not-equal";
    YearCondition["IsNotNull"] = "is-not-null";
    YearCondition["IsNumber"] = "is-number";
    YearCondition["IsOddNumber"] = "is-odd-number";
    YearCondition["IsThisYear"] = "is-this-year";
    YearCondition["IsYear"] = "is-year";
})(YearCondition || (YearCondition = {}));
