// @srclaunch/types/dist/applab/rules/condition/primitive/network.js
import { Condition } from '../index';
export var DataURICondition;
(function (DataURICondition) {
    DataURICondition["Contains"] = "contains";
    DataURICondition["IsDataURI"] = "is-data-uri";
    DataURICondition["IsEqual"] = "is-equal";
    DataURICondition["IsNotEqual"] = "is-not-equal";
    DataURICondition["IsNotNull"] = "is-not-null";
    DataURICondition["IsString"] = "is-string";
})(DataURICondition || (DataURICondition = {}));
export var DomainNameCondition;
(function (DomainNameCondition) {
    DomainNameCondition["Contains"] = "contains";
    DomainNameCondition["IsDomainName"] = "is-domain-name";
    DomainNameCondition["IsEqual"] = "is-equal";
    DomainNameCondition["IsNotEqual"] = "is-not-equal";
    DomainNameCondition["IsNotNull"] = "is-not-null";
    DomainNameCondition["IsString"] = "is-string";
})(DomainNameCondition || (DomainNameCondition = {}));
export var EmailCondition;
(function (EmailCondition) {
    EmailCondition["Contains"] = "contains";
    EmailCondition["IsEmailAddress"] = "is-email-address";
    EmailCondition["IsEqual"] = "is-equal";
    EmailCondition["IsInList"] = "is-in-list";
    EmailCondition["IsNotEqual"] = "is-not-equal";
    EmailCondition["IsNotInList"] = "is-not-in-list";
    EmailCondition["IsNotNull"] = "is-not-null";
    EmailCondition["IsString"] = "is-string";
})(EmailCondition || (EmailCondition = {}));
export var IPAddressCondition;
(function (IPAddressCondition) {
    IPAddressCondition["Contains"] = "contains";
    IPAddressCondition["IsEqual"] = "is-equal";
    IPAddressCondition["IsIPAddress"] = "is-ip-address";
    IPAddressCondition["IsInIPAddressRange"] = "is-in-ip-address-range";
    IPAddressCondition["IsInList"] = "is-in-list";
    IPAddressCondition["IsNotEqual"] = "is-not-equal";
    IPAddressCondition["IsNotInList"] = "is-not-in-list";
    IPAddressCondition["IsNotInIPAddressRange"] = "is-not-in-ip-address-range";
    IPAddressCondition["IsNotNull"] = "is-not-null";
    IPAddressCondition["IsString"] = "is-string";
})(IPAddressCondition || (IPAddressCondition = {}));
export var IPAddressRangeCondition;
(function (IPAddressRangeCondition) {
    IPAddressRangeCondition["IsEqual"] = "is-equal";
    IPAddressRangeCondition["IsIPAddressRange"] = "is-ip-address-range";
    IPAddressRangeCondition["IsInList"] = "is-in-list";
    IPAddressRangeCondition["IsNotEqual"] = "is-not-equal";
    IPAddressRangeCondition["IsNotInList"] = "is-not-in-list";
    IPAddressRangeCondition["IsNotInIPAddressRange"] = "is-not-in-ip-address-range";
    IPAddressRangeCondition["IsNotNull"] = "is-not-null";
    IPAddressRangeCondition["IsString"] = "is-string";
})(IPAddressRangeCondition || (IPAddressRangeCondition = {}));
export var PortCondition;
(function (PortCondition) {
    PortCondition["IsEqual"] = "is-equal";
    PortCondition["IsGreaterThan"] = "greater-than";
    PortCondition["IsGreaterThanOrEqual"] = "greater-than-or-equal";
    PortCondition["IsInteger"] = "is-integer";
    PortCondition["IsLessThan"] = "less-than";
    PortCondition["IsLessThanOrEqual"] = "less-than-or-equal";
    PortCondition["IsNotEqual"] = "is-not-equal";
    PortCondition["IsNotNull"] = "is-not-null";
})(PortCondition || (PortCondition = {}));
export var MACAddressCondition;
(function (MACAddressCondition) {
    MACAddressCondition["Contains"] = "contains";
    MACAddressCondition["IsEqual"] = "is-equal";
    MACAddressCondition["IsInList"] = "is-in-list";
    MACAddressCondition["IsMACAddress"] = "is-mac-address";
    MACAddressCondition["IsNotEqual"] = "is-not-equal";
    MACAddressCondition["IsNotInList"] = "is-not-in-list";
    MACAddressCondition["IsNotNull"] = "is-not-null";
    MACAddressCondition["IsString"] = "is-string";
})(MACAddressCondition || (MACAddressCondition = {}));
export var MagnetURICondition;
(function (MagnetURICondition) {
    MagnetURICondition["Contains"] = "contains";
    MagnetURICondition["IsEqual"] = "is-equal";
    MagnetURICondition["IsInList"] = "is-in-list";
    MagnetURICondition["IsMagnetURI"] = "is-magnet-uri";
    MagnetURICondition["IsNotEqual"] = "is-not-equal";
    MagnetURICondition["IsNotInList"] = "is-not-in-list";
    MagnetURICondition["IsNotNull"] = "is-not-null";
    MagnetURICondition["IsString"] = "is-string";
})(MagnetURICondition || (MagnetURICondition = {}));
export var MimeTypeCondition;
(function (MimeTypeCondition) {
    MimeTypeCondition["Contains"] = "contains";
    MimeTypeCondition["IsEqual"] = "is-equal";
    MimeTypeCondition["IsInList"] = "is-in-list";
    MimeTypeCondition["IsMimeType"] = "is-mime-type";
    MimeTypeCondition["IsNotEqual"] = "is-not-equal";
    MimeTypeCondition["IsNotInList"] = "is-not-in-list";
    MimeTypeCondition["IsNotNull"] = "is-not-null";
    MimeTypeCondition["IsString"] = "is-string";
})(MimeTypeCondition || (MimeTypeCondition = {}));
export var SlugCondition;
(function (SlugCondition) {
    SlugCondition["Contains"] = "contains";
    SlugCondition["IsEqual"] = "is-equal";
    SlugCondition["IsInList"] = "is-in-list";
    SlugCondition["IsNotEqual"] = "is-not-equal";
    SlugCondition["IsNotInList"] = "is-not-in-list";
    SlugCondition["IsNotNull"] = "is-not-null";
    SlugCondition["IsString"] = "is-string";
    SlugCondition["IsSlug"] = "is-slug";
})(SlugCondition || (SlugCondition = {}));
export var URLCondition;
(function (URLCondition) {
    URLCondition["Contains"] = "contains";
    URLCondition["IsEqual"] = "is-equal";
    URLCondition["IsInList"] = "is-in-list";
    URLCondition["IsNotEqual"] = "is-not-equal";
    URLCondition["IsNotInList"] = "is-not-in-list";
    URLCondition["IsNotNull"] = "is-not-null";
    URLCondition["IsString"] = "is-string";
    URLCondition["IsURL"] = "is-url";
})(URLCondition || (URLCondition = {}));
