// @srclaunch/types/dist/applab/rules/condition/primitive/number.js
import { Condition } from '../index';
export var NumberCondition;
(function (NumberCondition) {
    NumberCondition["IsAfter"] = "is-after";
    NumberCondition["IsAfterOrEqual"] = "is-after-or-equal";
    NumberCondition["IsBefore"] = "is-before";
    NumberCondition["IsBeforeOrEqual"] = "is-before-or-equal";
    NumberCondition["IsBetween"] = "is-between";
    NumberCondition["IsDecimal"] = "is-decimal";
    NumberCondition["IsDivisibleBy"] = "is-divisible-by";
    NumberCondition["IsEAN"] = "is-ean";
    NumberCondition["IsEIN"] = "is-ein";
    NumberCondition["IsEqual"] = "is-equal";
    NumberCondition["IsEvenNumber"] = "is-even-number";
    NumberCondition["IsFloat"] = "is-float";
    NumberCondition["IsGreaterThan"] = "greater-than";
    NumberCondition["IsGreaterThanOrEqual"] = "greater-than-or-equal";
    NumberCondition["IsInt"] = "is-integer";
    NumberCondition["IsISBN"] = "is-isbn";
    NumberCondition["IsISMN"] = "is-ismn";
    NumberCondition["IsISSN"] = "is-issn";
    NumberCondition["IsLatitude"] = "is-latitude";
    NumberCondition["IsLongitude"] = "is-longitude";
    NumberCondition["IsLessThan"] = "less-than";
    NumberCondition["IsLessThanOrEqual"] = "less-than-or-equal";
    NumberCondition["IsMACAddress"] = "is-mac-address";
    NumberCondition["IsNumber"] = "is-number";
    NumberCondition["IsNegativeNumber"] = "is-negative-number";
    NumberCondition["IsNotEqual"] = "is-not-equal";
    NumberCondition["IsNotNull"] = "is-not-null";
    NumberCondition["IsOddNumber"] = "is-odd-number";
    NumberCondition["IsPassportNumber"] = "is-passport-number";
    NumberCondition["IsPhoneNumber"] = "is-phone-number";
    NumberCondition["IsPort"] = "is-port";
    NumberCondition["IsPositiveNumber"] = "is-positive-number";
    NumberCondition["IsPostalCode"] = "is-postal-code";
    NumberCondition["IsSemanticVersion"] = "is-semantic-version";
    NumberCondition["IsSSN"] = "is-ssn";
    NumberCondition["IsTaxIDNumber"] = "is-tax-id-number";
    NumberCondition["IsUUID"] = "is-uuid";
    NumberCondition["IsVATIDNumber"] = "is-vat-id-number";
})(NumberCondition || (NumberCondition = {}));
export var FloatCondition;
(function (FloatCondition) {
    FloatCondition["IsEqual"] = "is-equal";
    FloatCondition["IsFloat"] = "is-float";
    FloatCondition["IsGreaterThan"] = "greater-than";
    FloatCondition["IsGreaterThanOrEqual"] = "greater-than-or-equal";
    FloatCondition["IsLessThan"] = "less-than";
    FloatCondition["IsLessThanOrEqual"] = "less-than-or-equal";
    FloatCondition["IsNotEqual"] = "is-not-equal";
    FloatCondition["IsNotNull"] = "is-not-null";
    FloatCondition["IsNumber"] = "is-number";
    FloatCondition["IsNumeric"] = "is-numeric";
})(FloatCondition || (FloatCondition = {}));
export var IntegerCondition;
(function (IntegerCondition) {
    IntegerCondition["IsEqual"] = "is-equal";
    IntegerCondition["IsInteger"] = "is-integer";
    IntegerCondition["IsGreaterThan"] = "greater-than";
    IntegerCondition["IsGreaterThanOrEqual"] = "greater-than-or-equal";
    IntegerCondition["IsLessThan"] = "less-than";
    IntegerCondition["IsLessThanOrEqual"] = "less-than-or-equal";
    IntegerCondition["IsNotEqual"] = "is-not-equal";
    IntegerCondition["IsNotNull"] = "is-not-null";
    IntegerCondition["IsNumber"] = "is-number";
    IntegerCondition["IsNumeric"] = "is-numeric";
})(IntegerCondition || (IntegerCondition = {}));
