// @srclaunch/types/dist/applab/rules/condition/primitive/standards.js
import { Condition } from '../index';
/**
 * Bank Identification Code
 *
 * @remarks
 * BIC means Bank Identification Code, or Bank Identifier Code.
 * It is an 8 to 11-character code that is used to identify a
 * specific bank when you make an international transaction.
 * */
export var BICCondition;
(function (BICCondition) {
    BICCondition["Contains"] = "contains";
    BICCondition["IsBIC"] = "is-bic";
    BICCondition["IsEqual"] = "is-equal";
    BICCondition["IsInList"] = "is-in-list";
    BICCondition["IsNotEqual"] = "is-not-equal";
    BICCondition["IsNotInList"] = "is-not-in-list";
    BICCondition["IsNotNull"] = "is-not-null";
    BICCondition["IsString"] = "is-string";
})(BICCondition || (BICCondition = {}));
/**
 * European Article Number
 *
 * @remarks
 * The International Article Number (also known as European
 * Article Number or EAN) is a standard describing a barcode
 * symbology and numbering system used in global trade to identify
 * a specific retail product type, in a specific packaging
 * configuration, from a specific manufacturer.
 *
 * The most commonly used EAN standard is the thirteen-digit EAN-13
 * */
export var EANCondition;
(function (EANCondition) {
    EANCondition["Contains"] = "contains";
    EANCondition["IsEAN"] = "is-ean";
    EANCondition["IsEqual"] = "is-equal";
    EANCondition["IsInList"] = "is-in-list";
    EANCondition["IsNotEqual"] = "is-not-equal";
    EANCondition["IsNotInList"] = "is-not-in-list";
    EANCondition["IsNotNull"] = "is-not-null";
    EANCondition["IsString"] = "is-string";
})(EANCondition || (EANCondition = {}));
/**
 * Employer Identification Number
 *
 * @remarks
 *
 * The Employer Identification Number (EIN), also known as the Federal
 * Employer Identification Number (FEIN) or the Federal Tax Identification
 * Number, is a unique nine-digit number assigned by the Internal Revenue
 * Service (IRS) to business entities operating in the United States for the
 * purposes of identification.
 */
export var EINCondition;
(function (EINCondition) {
    EINCondition["Contains"] = "contains";
    EINCondition["IsEIN"] = "is-ein";
    EINCondition["IsEqual"] = "is-equal";
    EINCondition["IsInList"] = "is-in-list";
    EINCondition["IsNotEqual"] = "is-not-equal";
    EINCondition["IsNotInList"] = "is-not-in-list";
    EINCondition["IsNotNull"] = "is-not-null";
    EINCondition["IsString"] = "is-string";
})(EINCondition || (EINCondition = {}));
/**
 * International Bank Account Number
 *
 * @remarks
 *
 * An IBAN, or international bank account number, is a standard
 * international numbering system developed to identify an overseas
 * bank account. The number starts with a two-digit country code, then
 * two numbers, followed by several more alphanumeric characters.
 */
export var IBANCondition;
(function (IBANCondition) {
    IBANCondition["Contains"] = "contains";
    IBANCondition["IsEqual"] = "is-equal";
    IBANCondition["IsIBAN"] = "is-iban";
    IBANCondition["IsInList"] = "is-in-list";
    IBANCondition["IsNotEqual"] = "is-not-equal";
    IBANCondition["IsNotInList"] = "is-not-in-list";
    IBANCondition["IsNotNull"] = "is-not-null";
    IBANCondition["IsString"] = "is-string";
})(IBANCondition || (IBANCondition = {}));
/**
 * International Standard Book Number
 *
 * @remarks
 *
 * ISBN is the acronym for International Standard Book Number. This
 * 10 or 13-digit number identifies a specific book, an edition of a
 * book, or a book-like product (such as an audiobook). Since 1970 each
 * published book has a unique ISBN. In 2007, assigned ISBNs changed
 * from 10 digits to 13.
 */
export var ISBNCondition;
(function (ISBNCondition) {
    ISBNCondition["Contains"] = "contains";
    ISBNCondition["IsEqual"] = "is-equal";
    ISBNCondition["IsISBN"] = "is-isbn";
    ISBNCondition["IsInList"] = "is-in-list";
    ISBNCondition["IsNotEqual"] = "is-not-equal";
    ISBNCondition["IsNotInList"] = "is-not-in-list";
    ISBNCondition["IsNotNull"] = "is-not-null";
    ISBNCondition["IsString"] = "is-string";
})(ISBNCondition || (ISBNCondition = {}));
/**
 * International Securities Identification Number
 *
 * @remarks
 * An ISIN is a 12-digit alphanumeric code that uniquely identifies a
 * specific security. The numbers are allocated by a country's respective
 * national numbering agency (NNA). An ISIN is not the same as the ticker
 * symbol, which identifies the stock at the exchange level.
 */
export var ISINCondition;
(function (ISINCondition) {
    ISINCondition["Contains"] = "contains";
    ISINCondition["IsEqual"] = "is-equal";
    ISINCondition["IsISIN"] = "is-isin";
    ISINCondition["IsInList"] = "is-in-list";
    ISINCondition["IsNotEqual"] = "is-not-equal";
    ISINCondition["IsNotInList"] = "is-not-in-list";
    ISINCondition["IsNotNull"] = "is-not-null";
    ISINCondition["IsString"] = "is-string";
})(ISINCondition || (ISINCondition = {}));
/**
 * International Standard Music Number
 *
 * @remarks
 * The International Standard Music Number or ISMN (ISO 10957) is a
 * thirteen-character alphanumeric identifier for printed music developed
 * by ISO.
 */
export var ISMNCondition;
(function (ISMNCondition) {
    ISMNCondition["Contains"] = "contains";
    ISMNCondition["IsEqual"] = "is-equal";
    ISMNCondition["IsISMN"] = "is-ismn";
    ISMNCondition["IsInList"] = "is-in-list";
    ISMNCondition["IsNotEqual"] = "is-not-equal";
    ISMNCondition["IsNotInList"] = "is-not-in-list";
    ISMNCondition["IsNotNull"] = "is-not-null";
    ISMNCondition["IsString"] = "is-string";
})(ISMNCondition || (ISMNCondition = {}));
/**
 * International Standard Serial Number
 *
 * @remarks
 * An International Standard Serial Number (ISSN) is an eight-digit
 * serial number used to uniquely identify a serial publication,
 * such as a magazine. The ISSN is especially helpful in distinguishing
 * between serials with the same title.
 */
export var ISSNCondition;
(function (ISSNCondition) {
    ISSNCondition["Contains"] = "contains";
    ISSNCondition["IsEqual"] = "is-equal";
    ISSNCondition["IsISSN"] = "is-issn";
    ISSNCondition["IsInList"] = "is-in-list";
    ISSNCondition["IsNotEqual"] = "is-not-equal";
    ISSNCondition["IsNotInList"] = "is-not-in-list";
    ISSNCondition["IsNotNull"] = "is-not-null";
    ISSNCondition["IsString"] = "is-string";
})(ISSNCondition || (ISSNCondition = {}));
/**
 * Tax Identification Number
 *
 * @remarks
 * A Tax Identification Number (TIN) is a nine-digit number used as a
 * tracking number by the Internal Revenue Service (IRS)
 */
export var TaxIDNumberCondition;
(function (TaxIDNumberCondition) {
    TaxIDNumberCondition["Contains"] = "contains";
    TaxIDNumberCondition["IsEqual"] = "is-equal";
    TaxIDNumberCondition["IsInList"] = "is-in-list";
    TaxIDNumberCondition["IsNotEqual"] = "is-not-equal";
    TaxIDNumberCondition["IsNotInList"] = "is-not-in-list";
    TaxIDNumberCondition["IsNotNull"] = "is-not-null";
    TaxIDNumberCondition["IsString"] = "is-string";
    TaxIDNumberCondition["IsTaxIDNumber"] = "is-tax-id-number";
})(TaxIDNumberCondition || (TaxIDNumberCondition = {}));
/**
 * Value Added Tax (VAT) Registration Number
 *
 * @remarks
 * A VAT registration number is a nine-digit number used to identify
 * a business or person in the European Union.
 */
export var VATCondition;
(function (VATCondition) {
    VATCondition["Contains"] = "contains";
    VATCondition["IsEqual"] = "is-equal";
    VATCondition["IsInList"] = "is-in-list";
    VATCondition["IsNotEqual"] = "is-not-equal";
    VATCondition["IsNotInList"] = "is-not-in-list";
    VATCondition["IsNotNull"] = "is-not-null";
    VATCondition["IsString"] = "is-string";
    VATCondition["IsVATIDNumber"] = "is-vat-id-number";
})(VATCondition || (VATCondition = {}));
