// @srclaunch/types/dist/applab/rules/condition/primitive/string.js
import { Condition } from '../index';
// TODO: Example the number of conditions a string can support
export var StringCondition;
(function (StringCondition) {
    StringCondition["Contains"] = "contains";
    StringCondition["HasNumberCount"] = "has-number-count";
    StringCondition["HasLowercaseCount"] = "has-lowercase-count";
    StringCondition["HasLetterCount"] = "has-letter-count";
    StringCondition["HasSpacesCount"] = "has-spaces-count";
    StringCondition["HasSymbolCount"] = "has-symbol-count";
    StringCondition["HasUppercaseCount"] = "has-uppercase-count";
    StringCondition["IsAlpha"] = "is-alpha";
    StringCondition["IsAlphanumeric"] = "is-alphanumeric";
    StringCondition["IsAscii"] = "is-ascii";
    StringCondition["IsBase64"] = "is-base-64";
    StringCondition["IsColor"] = "is-color";
    StringCondition["IsComplexEnough"] = "is-complex-enough";
    StringCondition["IsCreditCard"] = "is-credit-card";
    StringCondition["IsDataURI"] = "is-data-uri";
    StringCondition["IsDomainName"] = "is-domain-name";
    StringCondition["IsEmailAddress"] = "is-email-address";
    StringCondition["IsEthereumAddress"] = "is-ethereum-address";
    StringCondition["IsEAN"] = "is-ean";
    StringCondition["IsEIN"] = "is-ein";
    StringCondition["IsEqual"] = "is-equal";
    StringCondition["IsIBAN"] = "is-iban";
    StringCondition["IsHSLColor"] = "is-hsl-color";
    StringCondition["IsHexColor"] = "is-hex-color";
    StringCondition["IsHexadecimal"] = "is-hexadecimal";
    StringCondition["IsIdentityCardCode"] = "is-identity-card-code";
    StringCondition["IsIMEI"] = "is-imei";
    StringCondition["IsInList"] = "is-in-list";
    StringCondition["IsIPAddress"] = "is-ip-address";
    StringCondition["IsInIPAddressRange"] = "is-in-ip-address-range";
    StringCondition["IsISBN"] = "is-isbn";
    StringCondition["IsISIN"] = "is-isin";
    StringCondition["IsISMN"] = "is-ismn";
    StringCondition["IsISRC"] = "is-isrc";
    StringCondition["IsISSN"] = "is-issn";
    StringCondition["IsLanguage"] = "is-language";
    StringCondition["IsLatitude"] = "is-latitude";
    StringCondition["IsLongitude"] = "is-longitude";
    StringCondition["IsLengthEqual"] = "is-length-equal";
    StringCondition["IsLengthGreaterThan"] = "is-length-greater-than";
    StringCondition["IsLengthGreaterThanOrEqual"] = "is-length-great-than-or-equal";
    StringCondition["IsLengthLessThan"] = "is-length-less-than";
    StringCondition["IsLengthLessThanOrEqual"] = "is-length-less-than-or-equal";
    StringCondition["IsLicensePlateNumber"] = "is-license-plate-number";
    StringCondition["IsLowercase"] = "is-lowercase";
    StringCondition["IsOctal"] = "is-octal";
    StringCondition["IsMACAddress"] = "is-mac-address";
    StringCondition["IsMD5"] = "is-md5";
    StringCondition["IsMagnetURI"] = "is-magnet-uri";
    StringCondition["IsMarkdown"] = "is-markdown";
    StringCondition["IsMimeType"] = "is-mime-type";
    StringCondition["IsMonth"] = "is-month";
    StringCondition["IsNotInIPAddressRange"] = "is-not-in-ip-address-range";
    StringCondition["IsNotInList"] = "is-not-in-list";
    StringCondition["IsNotNull"] = "is-not-null";
    StringCondition["IsNotRegexMatch"] = "is-not-regex-match";
    StringCondition["IsNumber"] = "is-number";
    StringCondition["IsNumeric"] = "is-numeric";
    StringCondition["IsPassportNumber"] = "is-passport-number";
    StringCondition["IsPhoneNumber"] = "is-phone-number";
    StringCondition["IsPort"] = "is-port";
    StringCondition["IsPostalCode"] = "is-postal-code";
    StringCondition["IsProvince"] = "is-province";
    StringCondition["IsRegexMatch"] = "is-regex-match";
    StringCondition["IsSemanticVersion"] = "is-semantic-version";
    StringCondition["IsSlug"] = "is-slug";
    StringCondition["IsSSN"] = "is-ssn";
    StringCondition["IsState"] = "is-state";
    StringCondition["IsStreetAddress"] = "is-street-address";
    StringCondition["IsString"] = "is-string";
    StringCondition["IsTaxIDNumber"] = "is-tax-id-number";
    StringCondition["IsURL"] = "is-url";
    StringCondition["IsUUID"] = "is-uuid";
    StringCondition["IsUppercase"] = "is-uppercase";
    StringCondition["IsVATIDNumber"] = "is-vat-id-number";
    StringCondition["IsWeekday"] = "is-weekday";
    StringCondition["IsWeekend"] = "is-weekend";
    StringCondition["IsYear"] = "is-year";
})(StringCondition || (StringCondition = {}));
export var LongTextCondition;
(function (LongTextCondition) {
    LongTextCondition["Contains"] = "contains";
    LongTextCondition["IsAlpha"] = "is-alpha";
    LongTextCondition["IsAlphanumeric"] = "is-alphanumeric";
    LongTextCondition["IsInList"] = "is-in-list";
    LongTextCondition["IsMarkdown"] = "is-markdown";
    LongTextCondition["IsNotInList"] = "is-not-in-list";
    LongTextCondition["IsNumeric"] = "is-numeric";
    LongTextCondition["IsLowercase"] = "is-lowercase";
    LongTextCondition["IsString"] = "is-string";
    LongTextCondition["IsUppercase"] = "is-uppercase";
})(LongTextCondition || (LongTextCondition = {}));
